overlord/devicestate/devicemgr.go

```diff
@@ -1091,6 +1091,30 @@ func (m *DeviceManager) ensureCloudInitRestricted() error {
 	return nil
 }
 
+// hasInstallDeviceHook returns whether the gadget has an install-device hook.
+// It can return an error if the device has no gadget snap
+func (m *DeviceManager) hasInstallDeviceHook(model *asserts.Model) (bool, error) {
+	var hasInstallDeviceHook bool
+
+	gadgetInfo, err := snapstate.CurrentInfo(m.state, model.Gadget())
+	if err != nil {
+		return false, fmt.Errorf("device is seeded in install mode but has no gadget snap: %v", err)
+	}
+	hasInstallDeviceHook = (gadgetInfo.Hooks["install-device"] != nil)
+	return hasInstallDeviceHook, nil
+}
+
+func (m *DeviceManager) installDeviceHookTask(model *asserts.Model) *state.Task {
+	summary := i18n.G("Run install-device hook")
+	hooksup := &hookstate.HookSetup{
+		// TODO: what's a reasonable timeout for the install-device hook?
```

Comment:
I looked what the other install-device hook during install uses and it has the same TODO so this should be fine :)